bicytok/distance_metric_funcs.py

```diff
@@ -96,47 +117,142 @@ def KL_EMD_2D(
     assert targNorms.shape[0] != recAbundances.shape[0]
 
     row, col = np.tril_indices(
-        recAbundances.shape[1]
-    )  # Triangle indices, includes diagonal (k=0 by default)
+        recAbundances.shape[1], k=0
+    )  # k=0 includes diagonals which are 1D distances
     for rec1, rec2 in zip(row, col, strict=False):
+        if not calc_1D and rec1 == rec2:
+            continue
+
         targAbun1, targAbun2 = targNorms[:, rec1], targNorms[:, rec2]
         offTargAbun1, offTargAbun2 = offTargNorms[:, rec1], offTargNorms[:, rec2]
 
-        assert np.allclose(
-            targAbun1, recAbundances[targ, rec1] / np.mean(recAbundances[:, rec1])
-        )
+        if (
+            np.mean(recAbundances[:, rec1]) > 5
```

Comment:
I'd recommend doing these comparisons outside the loop as a broadcasted comparison, and then only looping over the matching indices. This will be a little cleaner. Let's still merge this, but then perhaps open an issue as a todo note.